packages/sanity/src/structure/panes/documentList/useDocumentSheetListStore.ts

```diff
@@ -0,0 +1,247 @@
+import {type ListenEvent, type ListenOptions} from '@sanity/client'
```

Comment:
High chances this will be deprecated by the stream of concurrent work in which we will have 1 unique listener for all documents and we could use a new hook that mimics `useEditState` but will take a list of ids.

```ts

function useEditStateList(publishedDocIds: string[], docTypeName: string): EditStateFor[] {
  const documentStore = useDocumentStore()
  return useMemoObservable(() => {
    return combineLatest(
      publishedDocIds.map((publishedDocId) =>
        documentStore.pair.editState(publishedDocId, docTypeName),
      ),
    )
  }, [documentStore.pair, publishedDocIds, docTypeName]) as EditStateFor[]
}
```